bokehjs/src/coffee/models/selections/selection.ts

```diff
@@ -0,0 +1,115 @@
+import {Model} from "../../model"
+import * as p from "core/properties"
+import {union, intersection} from "core/util/array"
+import {merge} from "core/util/object"
+import {Glyph, GlyphView} from "models/glyphs/glyph"
+
+export class Selection extends Model {
+  static initClass() {
+    this.prototype.type = "Selection";
+
+    this.define({
+      indices:         [ p.Array,   [] ],
+    });
+
+    this.internal({
+      final:             [ p.Boolean     ],
+      line_indices:      [ p.Array,   [] ],
+      selected_glyphs:   [ p.Array,   [] ],
+      get_view:          [ p.Any         ],
+      multiline_indices: [ p.Any,     {} ],
+    });
+  }
+
+  indices: number[]
+  final: boolean
+  line_indices: number[]
+  selected_glyphs: Glyph[]
+  get_view: () => GlyphView | null
+  multiline_indices: {[key: string]: number[]} // XXX: [key: number]?
+  [key: string]: any
+
+  initialize(): void {
+    super.initialize()
+
+    this['0d'] = {'glyph': null, 'indices': [], 'flag': false,
+                  'get_view': () => null}
+    this['2d'] = {'indices': {}}
+    this['1d'] = {'indices': this.indices}
+
+    this.get_view = () => null
+
+    this.connect(this.properties.indices.change, () =>
+      this['1d']['indices'] = this.indices)
+    this.connect(this.properties.line_indices.change, () => {
+      this['0d']['indices'] = this.line_indices
+      if(this.line_indices.length == 0)
+        this['0d'].flag = false
+      else
+        this['0d'].flag = true
+      })
+    this.connect(this.properties.selected_glyphs.change, () =>
+      this['0d'].glyph = this.selected_glyph)
+    this.connect(this.properties.get_view.change, () =>
+      this['0d'].get_view = this.get_view)
+    this.connect(this.properties.multiline_indices.change, ()=>
+      this['2d'].indices = this.multiline_indices)
+  }
+
+  get selected_glyph(): Glyph | null {
+    if(this.selected_glyphs.length > 0)
+      return this.selected_glyphs[0]
```

Comment:
The goal here is to eventually be able to get any of the glyphs that were hit as part of the selection. For now, this is just used as an implementation detail to let a Line GlyphRenderer know its Line was selected.